src/package.c

```diff
@@ -975,18 +975,70 @@ int apk_ipkg_add_script(struct apk_installed_package *ipkg,
 	return 0;
 }
 
-void apk_ipkg_run_script(struct apk_installed_package *ipkg,
-			 struct apk_database *db,
-			 unsigned int type, char **argv)
+static int run_script(char** argv, int dirfd, char* fn)
 {
+	int status;
+	pid_t pid;
 	static char * const environment[] = {
 		"PATH=/usr/sbin:/usr/bin:/sbin:/bin",
 		NULL
 	};
+
+	pid = fork();
+	if (pid == -1) {
+		apk_error("fork() failed with error str: %s", apk_error_str(errno));
+		return -1;
+	}
+	if (pid == 0) {
+		umask(0022);
+		if (fchdir(dirfd) == 0) {
+			if (fn == NULL)
+				execve(argv[0], argv, environment); /* pre/post hook */
+			else if (chroot(".") == 0)
+				execve(fn, argv, environment); /* apk "internal" script */
+			else
+				apk_error("chroot() failed for script with error str: %s", apk_error_str(errno));
+			apk_error("execve() failed for script with error str: %s", apk_error_str(errno));
+		} else
+			apk_error("change working dir failed with error str: %s", apk_error_str(errno));
+		exit(1); /* should not get here */
+	}
+	waitpid(pid, &status, 0);
+	if (!WIFEXITED(status) || WEXITSTATUS(status) != 0) {
+		apk_error("script exited with error %d", WEXITSTATUS(status));
+		return -1;
+	}
+	return 0;
+}
+
+int apk_run_hook_script(void *ctx, int dirfd, const char *file)
+{
+	char fn[PATH_MAX];
+	char *stage = (char *)ctx;
+	char *script_args[] = { "/bin/sh", "-c", "", NULL };
+
+	if ((apk_flags & (APK_NO_SCRIPTS | APK_SIMULATE)) != 0)
+		return 0;
+
+	snprintf(fn, sizeof(fn), "./" "%s", file);
+	script_args[2] = (char *) fn;
+
+	if (apk_verbosity >= 2)
+		apk_message("Calling apk %s-script: %s\n", stage, fn);
+
+	if (run_script(script_args, dirfd, NULL) != 0)
+		exit(1);
```

Comment:
exit is intentional, example:
alpine:~# apk add gawk

** etckeeper detected uncommitted changes in /etc prior to apk run
** Aborting apk run. Manually commit and restart.

ERROR: ./etckeeper: script exited with error 1

But possibly  apk_run_hook_script() for the post case should be called from an other  function i src/commit.c, e.g. last in apk_solver_commit() what do you think? 

The db is however committed now when we call post as the data from the database is used in the formatting of the commit message:

alpine:~# rm /etc/dirty 

alpine:~# apk add gawk
(1/1) Installing gawk (4.1.4-r0)
Executing busybox-1.26.1-r3.trigger
OK: 1160 MiB in 174 packages
etckeeper warning: hardlinked files could cause problems with git:
terminfo/v/vt200
terminfo/v/vt220
[master b430194] committing changes in /etc after apk run
 1 file changed, 1 insertion(+)

alpine:~# cd /etc/
alpine:/etc# git log -1
commit b430194019a4ffc98a7a3856e14e42ae20d672a9
Author: root <root@alpine.localdomain>
Date:   Thu Feb 9 08:42:39 2017 +0100

    committing changes in /etc after apk run
    
    Package changes:
    +gawk-4.1.4-r0